include/switcher.php

```diff
@@ -55,22 +57,31 @@ protected function get_elements( $links, $args ) {
 			$locale = $language->get_locale( 'display' );
 			$classes = array( 'lang-item', 'lang-item-' . $id, 'lang-item-' . esc_attr( $slug ) );
 			$url = null; // Avoids potential notice
-
-			if ( $current_lang = $links->curlang->slug == $slug ) {
-				if ( $args['hide_current'] && ! ( $args['dropdown'] && ! $args['raw'] ) ) {
-					continue; // Hide current language except for dropdown
-				} else {
-					$classes[] = 'current-lang';
+			$current_lang = ''; // Avoids potential notice
+
+			if ( 0 === $args['admin_render'] ) {
+				if ( $current_lang = $links->curlang->slug == $slug ) {
+					if ( $args['hide_current'] && ! ( $args['dropdown'] && ! $args['raw'] ) ) {
+						continue; // Hide current language except for dropdown
+					} else {
+						$classes[] = 'current-lang';
+					}
 				}
-			}
 
-			if ( null !== $args['post_id'] && ( $tr_id = $links->model->post->get( $args['post_id'], $language ) ) && $links->model->post->current_user_can_read( $tr_id ) ) {
-				$url = get_permalink( $tr_id );
-			} elseif ( null === $args['post_id'] ) {
-				$url = $links->get_translation_url( $language );
+				if ( null !== $args['post_id'] && ( $tr_id = $links->model->post->get( $args['post_id'], $language ) ) && $links->model->post->current_user_can_read( $tr_id ) ) {
+					$url = get_permalink( $tr_id );
+				} elseif ( null === $args['post_id'] ) {
+					$url = $links->get_translation_url( $language );
+				}
+			} else {
+				if ( isset( $args['lang'] ) && $args['lang'] == $slug ) {
+					if ( $args['hide_current'] && ! ( $args['dropdown'] && ! $args['raw'] ) ) {
+						continue; // Hide current language except for dropdown
+					}
+				}
 			}
 
-			if ( $no_translation = empty( $url ) ) {
+			if ( $no_translation = empty( $url ) && 0 === $args['admin_render'] ) {
```

Comment:
Does the class `no-translation` disturbs the block renderig? Otherwise I believe that we could remove the adidional test.

Reply:
I suppose that I had some problem when rendering the block in the admin context but I don't remember why and I can't reproduce the case anymore.
So I removed the test.